vscode: fix #76427 fallout

eadwu opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change
PR https://github.com/NixOS/nixpkgs/pull/76427
Original change https://github.com/NixOS/nixpkgs/commit/bb24fafbfad04cd23431de14176cdc581a906f66

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

eadwu committed:
vscode-utils: remove extra whitespace

eadwu committed:
vscode-with-extensions: fix extension directory

justin-gerhardt commented:
https://github.com/NixOS/nixpkgs/pull/76427 seems have completely undone the changes from https://github.com/NixOS/nixpkgs/commit/bb24fafbfad04cd23431de14176cdc581a906f66 (was it based on an old branch?). 
Your commit hardcodes the path as https://github.com/NixOS/nixpkgs/blob/0717848d7bb5adb01c7a9f293ed07676099de6f2/pkgs/applications/editors/vscode/with-extensions.nix#L73
but "vscode" isn't guaranteed in the current version
https://github.com/NixOS/nixpkgs/blob/319418f226ea214c3852c172e600fb8cd2aa79d9/pkgs/misc/vscode-extensions/vscode-utils.nix#L28


As far as I can tell reverting the changes from https://github.com/NixOS/nixpkgs/commit/bb24fafbfad04cd23431de14176cdc581a906f66 wasn't deliberate. Reapplying https://github.com/NixOS/nixpkgs/commit/bb24fafbfad04cd23431de14176cdc581a906f66 seems like a better course of action

eadwu committed:
vscode-extensions.vscode-utils: reapply some changes from bb24faf

https://github.com/NixOS/nixpkgs/commit/bb24fafbfad04cd23431de14176cdc581a906f66

eadwu commented:
I'll keep the `share/vscode/extensions` since it seems like it's used after the PR. https://github.com/NixOS/nixpkgs/blob/319418f226ea214c3852c172e600fb8cd2aa79d9/pkgs/misc/vscode-extensions/vscodeWithConfiguration.nix#L21

Was originally suggested here https://github.com/NixOS/nixpkgs/pull/71251#discussion_r377972428.

Though I'll probably eventually(?) take a deeper look to fixup what was added.
